ext/src/ruby_api/convert.rs

```diff
@@ -45,25 +55,31 @@ impl ToWasmVal for Value {
             ValType::ExternRef => {
                 let extern_ref_value = match self.is_nil() {
                     true => None,
-                    false => Some(ExternRef::new(OnStackValue::from(*self))),
+                    false => Some(ExternRef::new(RbExternRef::from(*self))),
                 };
 
                 Ok(Val::ExternRef(extern_ref_value))
             }
-            ValType::FuncRef => Ok(Val::FuncRef(Some(*self.try_convert::<&Func>()?.inner()))),
+            ValType::FuncRef => {
+                let func_ref_value = match self.is_nil() {
+                    true => None,
+                    false => Some(*self.try_convert::<&Func>()?.inner()),
+                };
+                Ok(Val::FuncRef(func_ref_value))
+            }
             ValType::V128 => err!("converting from Ruby to v128 not supported"),
         }
     }
 }
 
-struct OnStackValue(Value);
```

Comment:
Renamed this because externrefs aren't necessarily on the stack anymore with table.